### ISSUE TITLE
Globs are not working for $ istanbul instrument -x "**glob**"

### ISSUE THREAD

**Issue by ORESoftware** [istanbuljs/nyc#596]
I am trying to do this:

```
public
    /lib     # this is library code (angular etc.), I want to ignore this dir
    /pages   # this is my code
```
    
Here is the command I use:

`istanbul instrument public --output public-coverage  -x "public/lib/**/*"`

Unfortunately, the public/lib directory still gets instrumented, which is a bummer because it contains a ton of files that are 3rd party libraries which I really do not want to instrument.

I have tried every glob combination under the sun and it's not working - the lib directory and its contents never get ignored, and they are always copied over to public-coverage and instrumented.

I consistently have problems with getting globs to behave as desired...it's not just this library. Not sure why that is, but maybe people should consider other ways of ignoring dirs, etc.

**Comment by ORESoftware** [comment]
bump pls

**Comment by JaKXz** [comment]
@ORESoftware I think you're using the wrong executable, but you can configure `nyc` to ignore directories like this: https://github.com/istanbuljs/nyc#excluding-files

Please share your repo or a sample reproduction with `nyc`.  I'd go as far as to unofficially say that the `istanbul` CLI is no longer supported, since `nyc` has replaced it. You can see the docs on the `instrument` command here: https://istanbul.js.org/docs/advanced/instrument/

**Comment by ORESoftware** [comment]
Thanks! I wonder why nyc didn't choose to stick with original CLI:

before:

`istanbul instrument ./lib -o ./output`

after?

`nyc instrument ./lib ./output`

**Comment by ORESoftware** [comment]
Unfortunately this doesn't work:

```
"nyc": {
    "exclude": [
      "public/lib/**/*"
    ]
  }
```

nyc still attempts to instrument the public/lib contents

this is driving me INSANE

**Comment by bcoe** [comment]
@ORESoftware could you please provide a link to a repository that is running into this issue?

Here's a unit test we have for logic very similar to what you're attempting:

https://github.com/istanbuljs/istanbuljs/blob/master/packages/test-exclude/test/test-exclude.js#L219

We're not doing anything exciting with our `test-exclude` logic, it simply runs a glob with a few nice defaults applied, e.g., always excluding `node_modules`.

If you are having trouble excluding `/public/lib` my guess is that there's something else odd happening, such as the path getting munged at some point during the instrumentation process.

**Comment by coreyfarrell** [comment]
Fix being worked on in #1007
